### ISSUE TITLE
nvm is not compatible with the npm config "prefix"...

### ISSUE THREAD

**Issue by Edditoria** [platformio/platformio-atom-ide-terminal#576]
When open a new terminal, the following error will occur:

```
nvm is not compatible with the npm config "prefix" option: currently set to "/usr/local"
Run `npm config delete prefix` or `nvm use --delete-prefix v8.11.1 --silent` to unset it.
```

I've found that the problem may be the `$PATH` that is different with Mac's Terminal.

```
# echo $PATH in Terminal:

/Users/Edditoria/.pyenv/shims:/Users/Edditoria/.pyenv/bin:/Users/Edditoria/.rbenv/shims:/Users/Edditoria/.nvm/versions/node/v8.11.1/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin

# echo $PATH in platformio-atom-ide-terminal:

/Users/Edditoria/.pyenv/shims:/Users/Edditoria/.pyenv/bin:/Users/Edditoria/.rbenv/shims:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:
/opt/X11/bin:/Users/Edditoria/.pyenv/shims:/Users/Edditoria/.pyenv/bin:/Users/Edditoria/.rbenv/shims
```

Other than `nvm` is missing, you can see that `.pyenv` and `.rbenv` are duplicated in platformio-atom-ide-terminal.

What should I do to spot the problem? And how to solve it? I really want to understand why it happens.

Thanks.

**Comment by the-j0k3r** [comment]
So I assume if you use an external terminal this issue isnt present?

**Comment by Edditoria** [comment]
Yes. No problem in Terminal

For the `$PATH`, it is set in `.bash_profile`:

```
# nvm for npm
export NVM_DIR=$HOME/.nvm
if [[ -s $(brew --prefix nvm)/nvm.sh ]]; then
	source $(brew --prefix nvm)/nvm.sh
fi

# rbenv for ruby
if which rbenv > /dev/null; then eval "$(rbenv init -)"; fi

# pyenv for python
export PYENV_ROOT="$HOME/.pyenv"
export PATH="$PYENV_ROOT/bin:$PATH"
if which pyenv > /dev/null; then eval "$(pyenv init -)"; fi
```

It seems that path of pyenv and rbenv are exported twice and doesn't export nvm.

Many thanks.

**Comment by the-j0k3r** [comment]
@Edditoria 

Thank you for that information.

Sadly, [this project is no longer under active development](https://github.com/platformio/platformio-atom-ide-terminal/issues/543) its entered a slow maintenance mode and there are quite a few issues outstanding regarding platformio-ide-terminal internals some users experience that are unresolved.

The project needs contributors to help chip away at these issues, which means anything like your issue that can be resolved by using external terminals is very likely not on anyone's radar except the people who experience it.

I recently came across a similar issue when running `npm install` from within Atom platformio-ide-terminal and failed to install all packages and setup the environment as  defined inproject  `package.json`, but after using an external terminal to do that initial setup platformio-ide-terminal was able to run OK

TL:DR;
If you know of any fixes that you can share or are able to submit a PR to fix this then that would be the faster route out of your situation.

**Comment by the-j0k3r** [comment]
I think this is related to #546 if not an actual duplicate of, the symptoms are pretty much the same.

we wait on #585 being merged which will fix these issues.

**Comment by the-j0k3r** [comment]
try again with platformio-atom-ide-terminal 2.9.0 should be all fixed.

**Comment by the-j0k3r** [comment]
This should be fixed now with 2.9.0 all similar issues have disappeared =)
